packages/next/src/server/lib/router-server.ts

```diff
@@ -142,6 +146,57 @@ export async function initialize(opts: {
     require('./render-server') as typeof import('./render-server')
 
   const requestHandlerImpl: WorkerRequestHandler = async (req, res) => {
+    if (
+      !opts.minimalMode &&
+      config.i18n &&
+      config.i18n?.localeDetection !== false
```

Comment:
I don't think that we need the optional chaining here since you've just checked that the `config.i18n` is present